File: Java/Sorting Algo/SelectionSort.java
class SelectionSort {
    public void sort(int[] nums) {
        int n = nums.length;
        for(int i=0; i<n-1; i++){
            int min = i;
            for(int j=i+1; j<n; j++){
                if(nums[j] < nums[min])
                    min = j;
            }
            int temp = nums[i];
            nums[i] = nums[min];
            nums[min] = temp;
        }
    }
}

File: DSA/Notes and examples/selectionsort.cpp
//find min value element of an unsorted array and swap it with first element
#include <iostream>
using namespace std;

void selectionSort()
{
  int i,j,temp;
  int arr[]={10,2,3,4,5};
  for(i=0;i<4;i++)
  {
      for(j=i+1;j<5;j++)
      {
          if(arr[j]<arr[i])
          {
              //swapping	
              temp=arr[j];
              arr[j]=arr[i];
              arr[i]=temp;
          }
      }
  }
  cout<<"sorted array:\n";
  for(i=0;i<5;i++)
  cout<<arr[i]<<" \n";
}


int main() {
    int arr[]={10,2,3,4,5};
    // int i,j,temp;
    cout<<"given array is:\n";
    for(int i=0;i<5;i++)
    cout<<arr[i]<<" \n";
    selectionSort();
    return 0;
}
